### PR TITLE
Adds missing sig algs

### PR DESCRIPTION
**Pull request by christianpaquin**
Adds missing sig algs from OQS spring release; addresses issue #194.

##### Checklist
- [x] Update README
- [x] Dilithium3 (missing hybrid)
- [x] Add falcon512, falcon1024
- [x] Add MQDSS-31-48

The following algs were not added: they caused failures that would need to be investigated:

- MQDSS-31-64
- Rainbow
- SPHINCS

### CODE REVIEW

**Comment by christianpaquin** [comment]
@xvzcf, I updated the `generate.yml` file to add falcon, but `python3 oqs-template/generate.py` does nothing. Are my edits ok?

File: oqs-template/generate.yml [hunk 1]
```diff
@@ -309,6 +309,35 @@ sigs:
...
+        oqs_meth: "OQS_SIG_alg_falcon_1024"
+        security: 256
+        oid: "1.3.9999.3.4"
+        mix_with: [{"name": "p521",
+                    "pretty_name": "ECDSA p521",
+                    "oid": "1.3.9999.3.5"},
+                   {"name": "rsa4096",
+                    "pretty_name": "RSA4096",
+                    "oid": "1.3.9999.3.6"}]
+    # iso (1)
```
Comment (xvzcf) [inline oqs-template/generate.yml:354]: I think you forgot to add a dash before line 339.

**Comment by christianpaquin** [comment]
@xvzcf, after fixing the `oqs-template/generate.yml` file and running `make generate_crypto_objects`, the compilation fails. Falcon is missing from `include/crypto/asn1.h`, I can't find the relevant file to modify in `oqs-template`.

**Comment by xvzcf** [comment]
[This patch](https://github.com/open-quantum-safe/openssl/files/4678484/templating_fix.txt) should fix the templating issues.

**Comment by christianpaquin** [comment]
We don't support L5 hybrid schemes yet (falcon1024); I need to enable it to fix the failure.

**Comment by christianpaquin** [comment]
@dstebila, @xvzcf; this is now failing because ledacrypt ([PR756](https://github.com/open-quantum-safe/liboqs/pull/756)) has been removed from OQS. Is one of you in the process of creating a PR to remove it from OpenSSL as well? If not, I'll do it now.

**Comment by xvzcf** [comment]
Ledacrypt has been removed from OpenSSL as well, a rebase should probably fix the failures.

**Comment by dstebila** [comment]
Yes, that was merged yesterday with #204.

**Comment by christianpaquin** [comment]
@xvzcf, this is ready for review.

**Comment by xvzcf** [comment]
A couple things:

- We were able to get SPHINCS and Rainbow to work after increasing some size limits that are given in the [prototyping post-quantum cryptography paper](https://s3.amazonaws.com/files.douglas.stebila.ca/files/research/papers/NISTPQC-CroPaqSte19.pdf). I'm not sure if all the changes we made are present in the mainline branch, but if they aren't, I'm assuming its an easy fix to bump those limits up?

- Aren't the `picnic2` variants missing?

- I'm not sure off the top of my head what is causing MQDSS-31-64 to fail. It was working fine in BoringSSL.

**Comment by baentsch** [comment]
@xvzcf FYI, I have the NIST security level dependent KEM generator also nearly complete -- but as it requires `make generate_crypto_objects` to be run -- as does this PR, we'd have _many_ conflicts if I submit before this is merged... So it'd be great if you could review and merge this PR before my morning so that I can either rebase or resolve conflicts until you get in tomorrow so I can do the PR and we (hopefully) make our 'end-of-week' deadline for freezing oqs-openssl. No pressure  :-)

**Review by xvzcf — APPROVED** [review]

**Pull request merged**